src/Tester/MembershipTests/LivenessTests.cs

```diff
@@ -134,27 +124,79 @@ protected static void DoClassInitialize()
 
             logger.Info("\n\n\n\nAbout to start sending msg to grain again\n\n\n");
 
-            for (int i = 0; i < grains.Count; i++)
+            for (int i = 0; i < numGrains; i++)
             {
-                long key = i + 1;
-                ILivenessTestGrain g1 = grains[i];
-                Assert.AreEqual(key, g1.GetPrimaryKeyLong());
-                Assert.AreEqual(key.ToString(CultureInfo.InvariantCulture), await g1.GetLabel());
-                await LogGrainIdentity(logger, g1);
+                await SendTraffic(i + 1);
             }
 
             for (int i = numGrains; i < 2 * numGrains; i++)
             {
-                long key = i + 1;
-                ILivenessTestGrain g1 = GrainClient.GrainFactory.GetGrain<ILivenessTestGrain>(key);
-                grains.Add(g1);
-                Assert.AreEqual(key, g1.GetPrimaryKeyLong());
-                Assert.AreEqual(key.ToString(CultureInfo.InvariantCulture), await g1.GetLabel());
-                await LogGrainIdentity(logger, g1);
+                await SendTraffic(i + 1);
             }
             logger.Info("======================================================");
         }
 
+        protected async Task Do_Liveness_OracleTest_3()
```

Comment:
This is a third test scenario in the base class (`LivenessTestsBase `). then the derived test classes (Azure, ZK, SQL, Grain) will call this method under the right MBR type.